Cookbook pre-review checklist template (#451)
ProjectPythia/projectpythia.github.io: portal/cookbook-tasklist.md

# Cookbook Author Checklist Template

Project Pythia maintainers will provide this checklist to Cookbook authors to help get Cookbooks ready for publication.

```md
**I am your Cookbook advocate, and my GitHub handle is <my-username>. Please tag me in this issue with any problems getting your Cookbook published!**

> Once we've marked this entire checklist, [click here to open an issue on ProjectPythia/cookbook-gallery to publish your Cookbook!](https://github.com/ProjectPythia/cookbook-gallery/issues/new?assignees=ProjectPythia%2Feducation&labels=content%2Ccookbook-gallery-submission&projects=&template=update-cookbook-gallery.yaml&title=Update+Gallery+with+new+Cookbook)

Author: My thought was that these are created as soon as an "advocate" is assigned to the Cookbook, and can help guide development such that when someone clicks "Submit a new Cookbook", they really are ready to submit.

I'd bet refactoring our Cookbook guide to be more of a checklist and then having this be more trim and only opened after submission would accomplish the same goal, if we prefer that approach.

> Once we've marked this entire checklist, [click here to open an issue on ProjectPythia/cookbook-gallery to publish your Cookbook!](https://github.com/ProjectPythia/cookbook-gallery/issues/new?assignees=ProjectPythia%2Feducation&labels=content%2Ccookbook-gallery-submission&projects=&template=update-cookbook-gallery.yaml&title=Update+Gallery+with+new+Cookbook)

Reviewer: Sorry, I must be missing something. What triggers the assignment of an advocate? I thought this happened when the author clicks "submit a new cookbook"

> Once we've marked this entire checklist, [click here to open an issue on ProjectPythia/cookbook-gallery to publish your Cookbook!](https://github.com/ProjectPythia/cookbook-gallery/issues/new?assignees=ProjectPythia%2Feducation&labels=content%2Ccookbook-gallery-submission&projects=&template=update-cookbook-gallery.yaml&title=Update+Gallery+with+new+Cookbook)

Reviewer 2: It looks like that is done manually? I don't see an action that would do this.

> Once we've marked this entire checklist, [click here to open an issue on ProjectPythia/cookbook-gallery to publish your Cookbook!](https://github.com/ProjectPythia/cookbook-gallery/issues/new?assignees=ProjectPythia%2Feducation&labels=content%2Ccookbook-gallery-submission&projects=&template=update-cookbook-gallery.yaml&title=Update+Gallery+with+new+Cookbook)

Reviewer: How does anyone know to do it manually? Is the Pythia team even aware of the cookbook under development before the creator clicks "submit a new cookbook"?

> Once we've marked this entire checklist, [click here to open an issue on ProjectPythia/cookbook-gallery to publish your Cookbook!](https://github.com/ProjectPythia/cookbook-gallery/issues/new?assignees=ProjectPythia%2Feducation&labels=content%2Ccookbook-gallery-submission&projects=&template=update-cookbook-gallery.yaml&title=Update+Gallery+with+new+Cookbook)

Author: The "Submit a New Cookbook" link is for filling out our issue template on the gallery. If the machinery is still working, that issue kicks off our automation to open a PR to update the gallery. We placed this button at the end of the Cookbook Guide because historically we tried to guide folks through the full creation process before requesting publication to the gallery.

We don't currently have any automation for assigning advocates to Cookbooks. This has all been manual coming out of hackathon interactions, Discourse discussion, or personal communication, and we have no dedicated process for these assignments or tracking them. We are usually made fully aware of Cookbook developments via Discourse posts or by the time someone is looking to transfer a repo to the Project Pythia org, per the Cookbook Guide instructions.

We can inject this checklist whenever and wherever it's more useful, and I'll make the specifics of the wording match. Then we should build out the advocate and reviewer process structure and tooling from there. For now the primary goal I have with this PR is to get currently languishing Cookbooks cared for and to be step 1 towards the broader review process.

> Once we've marked this entire checklist, [click here to open an issue on ProjectPythia/cookbook-gallery to publish your Cookbook!](https://github.com/ProjectPythia/cookbook-gallery/issues/new?assignees=ProjectPythia%2Feducation&labels=content%2Ccookbook-gallery-submission&projects=&template=update-cookbook-gallery.yaml&title=Update+Gallery+with+new+Cookbook)

Reviewer 3: I added some suggested text above advising folks to contact the Pythia team to be assigned an Advocate and initiate the review process, if they haven't already. I think this should typically happen *before* opening the PR to include the cookbook on the gallery, though it shouldn't break anything if someone pushes the button and opens the PR earlier in the process.

I agree with @dcamron that we should view this as a step toward a more formal review process, and the primary audience right now is the group of people who participated in the 2024 hackathon.

At our current stage, I don't think we should be trying to build automation around review or advocate assignments. We can handle these assignments manually easily enough. I think we need to prioritize giving people the information they need to successfully publish their cookbooks.

---

- [ ] **Confirm you’ve followed the entire Project Pythia [Cookbook Guide](https://projectpythia.org/cookbook-guide.html)**.

> Take note especially of the **Develop your cookbook**, **Authorship and the CITATION.cff file**, **Gallery tags**, and **Generate a DOI** sections. **Save the step of making a “release” of your cookbook as the last step of this checklist.**

Reviewer 3: I advocate leaving the entire DOI and release generation until the end of the review process. The "release" should signify some level of "finished" product IMO. I also think the complexity of linking to Zenodo may turn some folks off. The Advocate should help shepherd this through at the end of the review process.

Here's my specific suggestion:
```suggestion
Take note especially of the [Develop your cookbook](https://projectpythia.org/cookbook-guide.html#develop-your-cookbook), [Authorship and the CITATION.cff file](https://projectpythia.org/cookbook-guide.html#authorship-and-the-citation-cff-file), and [Gallery tags](https://projectpythia.org/cookbook-guide.html#gallery-tags) sections. **Save the [Generate a DOI](https://projectpythia.org/cookbook-guide.html#generate-a-doi) step as the last step of this checklist.**
```

    - Click the Actions tab for your repository.
    ![GitHub Actions tab link](https://raw.githubusercontent.com/ProjectPythia/projectpythia.github.io/main/portal/_static/images/1-actions.png "Actions Tab")
    - Highlight the `trigger-replace-links` action in the workflows sidebar.
    ![Sidebar list of workflows in GitHub Actions](https://raw.githubusercontent.com/ProjectPythia/projectpythia.github.io/main/portal/_static/images/2-actions-list.png "List of Workflows")
    - On the right-hand side of the page, Run workflow > on Branch: Main.
    ![Prompt for manually running the workflow](https://raw.githubusercontent.com/ProjectPythia/projectpythia.github.io/main/portal/_static/images/3-run-workflow.png "Run Workflow prompt")
- [ ] **Fill in all template sections of your README**.
This will serve as your Cookbook homepage
    - [ ] Title
    - [ ] Cookbook description (brief, under title)
    - [ ] Cookbook Motivation - use this as an opportunity to tell us how your Cookbook fits in the broader learning ecosystem. Who should use this book? Why is it needed? Where does its content begin and end relative to existing resources?

>     - [ ] Structure - this section is an optional roadmap for Cookbooks with more complicated structure. If you only have one main body of content that progresses linearly, you can leave this out.

Author: I decided this should be optional. Thoughts?

- [ ] **Confirm that your Cookbook is successfully building and publishing via GitHub Actions.**
This can be seen in individual Pull Requests as green checkmarks ✅ for important automation, especially the trigger-book-build action. You can also view a historical list of any of these Actions in the Actions tab at the top of your Cookbook repository. Check out nightly-build and trigger-book-build of PRs, then the build/build jobs to identify code errors. Please comment in this thread if you have issues identifying the source of any build and publishing failures your Cookbook has. Common failures include
    - Incorrectly specified environment.yml
    - trigger-link-check will fail if links in your content can not be resolved. We can help ignore links that are broken even if they work on manual clicks.
    - Code errors in your notebooks themselves
- [ ] **Identify a Maintainer team via GitHub handle(s) in this thread.**

> This can be one or more people with availability to check in on this Cookbook, issue fixes to broken content, or with a vision for the future development of the Cookbook.

Reviewer 3: ```suggestion
This can be one or more people with availability to check in on this Cookbook, issue fixes to broken content, or with a vision for the future development of the Cookbook. This is typically (but not necessarily) one of the primary authors of the Cookbook.  
```

Reviewer 3: Just wanted to add emphasis here, authoring a Cookbook carries with it some expectation of sticking around to help with maintenance.

> - [ ] **Finally, release your Cookbook!**

Reviewer 3: This goes along with my suggestion above to defer DOI generation until the end:
```suggestion
- [ ] **Link your Cookbook repo to Zenodo for DOI generation**
Follow steps 1-4 under [Generate a DOI](https://projectpythia.org/cookbook-guide.html#generate-a-doi) in the Cookbook Guide.
- [ ] **Finally, release your Cookbook!**
```

Author: On the guide, we have "release your cookbook" as a step included in generating the DOI. I will untangle this wording a little bit.
